clippy_lints/src/allow_attribute.rs

```diff
@@ -0,0 +1,93 @@
+use ast::{AttrStyle, MetaItemKind};
+use clippy_utils::{diagnostics::span_lint_and_sugg, source::snippet};
+use rustc_ast as ast;
+use rustc_errors::Applicability;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::{symbol::Ident, BytePos};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Detects uses of the `#[allow]` attribute and suggests to replace it with the new `#[expect]` attribute implemented by `#![feature(lint_reasons)]` ([RFC 2383](https://rust-lang.github.io/rfcs/2383-lint-reasons.html))
+    /// ### Why is this bad?
+    /// Using `#[allow]` isn't bad, but `#[expect]` may be preferred as it lints if the code **doesn't** produce a warning.
+    /// ### Example
```

Comment:
What do you think of this description? I think it's a bit clearer :)
```suggestion
    /// Detects uses of the `#[allow]` attribute and suggests replacing it with
    /// the `#[expect]` (See [RFC 2383](https://rust-lang.github.io/rfcs/2383-lint-reasons.html))
    ///
    /// The expect attribute is still unstable and requires the `lint_reasons`
    /// on nightly. It can be enabled by adding `#![feature(lint_reasons)]` to
    /// the crate root.
    ///
    /// This lint only warns outer attributes (`#[allow]`), as inner attributes
    /// (`#![allow]`) are usually used to enable or disable lints on a global scale. 
    /// 
    /// ### Why is this bad?
    /// 
    /// `#[expect]` attributes suppress the lint emission, but emit a warning, if
    /// the expectation is unfulfilled. This can be useful to be notified when the
    /// lint is no longer triggered.
    ///
    /// ### Example
```

Reply:
Fixed